### PR TITLE
feat: update client to use google analytics 4 and migrate to google identity services

### PR DESCRIPTION
**Pull request by samshara**
Configuration changes
replace REACT_APP_GA_TRACKING_ID with REACT_APP_GA_MEASUREMENT_ID

- Addresses 

    https://github.com/the-deep/client/issues/2573

    https://github.com/the-deep/client/issues/2610

- Depends on https://github.com/the-deep/deeper/pull/980

## Changes
* Update to use google-analytics v4
* Remove `react-ga` library
* migrate to google identity services

## Changes to configurations
* replace REACT_APP_GA_TRACKING_ID with REACT_APP_GA_MEASUREMENT_ID

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

**Review by AdityaKhatri — APPROVED** [review]

File: app/hooks/useScript.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
```
Comment (tnagorra) [inline app/hooks/useScript.ts:2]: Where did we get this hook from?

File: package.json [hunk 2]
```diff
@@ -65,7 +65,6 @@
         "mapbox-gl": "^2.4.1",
         "react": "^17.0.2",
         "react-dom": "^17.0.2",
-        "react-ga": "^3.3.0",
```
Comment (tnagorra) [inline package.json:68]: Why not use https://github.com/codler/react-ga4

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/hooks/useScript.ts (hunk 1 above)
Comment (samshara) [reply app/hooks/useScript.ts:2]: https://usehooks-ts.com/react-hook/use-script slightly modified from here.

File: package.json (hunk 2 above)
Comment (samshara) [reply package.json:68]: @tnagorra this library is a new one. previously it was https://github.com/PriceRunner/react-ga4 which was unmaintained. I will integrate the mentioned library.

File: app/index.html [hunk 3]
```diff
@@ -72,11 +72,14 @@
...
         </script>
         <!-- End of deephelp Zendesk Widget script -->
-
-        <!-- Start of Dropbox and Google Drive -->
-        <script type="text/javascript" src="https://apis.google.com/js/api.js"></script>
+        <% if (process.env.REACT_APP_GA_MEASUREMENT_ID) { %>
+            <!-- Start of Google tag (gtag.js) -->
+            <script async src="https://www.googletagmanager.com/gtag/js?id=<%=process.env.REACT_APP_GA_MEASUREMENT_ID%>"></script>
+            <!-- End of Google tag (gtag.js) -->
+        <% } %>
```
Comment (tnagorra) [inline app/index.html]:
```suggestion
        <!-- Start of Google tag (gtag.js) -->
        <% if (process.env.REACT_APP_GA_MEASUREMENT_ID) { %>
            <script async src="https://www.googletagmanager.com/gtag/js?id=<%=process.env.REACT_APP_GA_MEASUREMENT_ID%>"></script>
        <% } %>
        <!-- End of Google tag (gtag.js) -->
```

File: app/Base/index.tsx [hunk 4]
```diff
@@ -57,13 +57,17 @@ setMapboxToken(mapboxToken);
 if (sentryConfig) {
     init(sentryConfig);
 }
-
-ReactGA.initialize(trackingId, gaConfig);
-browserHistory.listen((location) => {
-    const page = location.pathname ?? window.location.pathname;
-    ReactGA.set({ page });
-    ReactGA.pageview(page);
```
Comment (tnagorra) [inline app/Base/index.tsx:65]: Do we not need to call ReactGA.pageview?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — APPROVED** [review]

File: app/index.html (hunk 3 above)
Comment (samshara) [reply app/index.html]: done

File: app/Base/index.tsx (hunk 4 above)
Comment (samshara) [reply app/Base/index.tsx:65]: done

**Pull request merged**